### PR TITLE
Added Redis Adapter for Cachified

### PR DESCRIPTION
**Pull request by mannyv123**
This pull request aims to bring the cachified redis adapter out of `@epic-web/cachified` and into it's own npm package. The following is included in this PR:

- Added a Redis 4 adapter for Cachified
- Updated unit tests to ensure adapter functionality

**Request for Review:**

Please carefully review these changes and provide feedback. Key areas to focus on:

- Adapter functionality and integration with Cachified
- Code quality and adherence to best practices
- Effectiveness of release automation

Thank you for your time and consideration!

### CODE REVIEW

File: src/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import {redisCacheAdapter} from './index'
```
Comment (tearingItUp786) [inline src/index.spec.ts:1]: 🪡 Nitpick: can just do `./` no?

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    value: string,
+    options?: {
+      EXAT: number
+    },
+  ): Promise<string | null>
+  get(key: string): Promise<string | null>
+  del(key: string): Promise<unknown>
+}
+
+// Adapter function
```
Comment (tearingItUp786) [inline src/index.ts:20]: This comment is unnecessary!

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            }
+          : undefined,
+      )
+    },
+    async get(key) {
+      const value = await redisCache.get(key)
+      if (value == null) {
+        return null
+      }
+      return JSON.parse(value)
```
Comment (tearingItUp786) [inline src/index.ts:44]: I would leave a comment here saying something along the lines that parse can throw an error and the expectation is that the user of the adapter catches it!

**Comment by tearingItUp786** [comment]
Nice, good job! 🥳 . Some minor tweaks needed here and there and we can then publish version 1 (squash it all into one feat commit)!

After we have version 1, let's install it on manjot.dev and take it out for a spin!

File: src/index.spec.ts (hunk 1 above)
Comment (mannyv123) [reply src/index.spec.ts:1]: updated

File: src/index.ts (hunk 2 above)
Comment (mannyv123) [reply src/index.ts:20]: removed comment

File: src/index.ts (hunk 3 above)
Comment (mannyv123) [reply src/index.ts:44]: added comment

**Comment by mannyv123** [comment]
Resolved comments and added `README.md` file and `.all-contributorsrc` file

File: README.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+[@epic-web/cachified](https://github.com/epicweb-dev/cachified) originally
+contained all adapters as a part of the same package. However, this can cause an
+issue when an adapter is updated and it becomes a breaking change release for
+the entire `@epic-web/cachified` package. This new npm package helps avoid this
+for the `redis` adapter and moves the adapter to its own package.
+
+## Install
+
+```bash
+npm install cachified-redis-adapter
```
Comment (tearingItUp786) [inline README.md:19]: aren't you using pnpm?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+const redisCache = redisCacheAdapter(redisClient)
+
+// usage with cachified
+return cachified({
+  key: `some-cache-key`,
+  // use the cache we defined above
+  cache: redisCache,
+  getFreshValue: async () => {}, // some function to get fresh values
+  // other cachified optoins
+})
```
Comment (tearingItUp786) [inline README.md:41]: Add a section linking to `cachified-redis-json` if people want to use that instead.

Comment (mannyv123) [reply README.md]: added link under **Usage** section of readme

**Comment by mannyv123** [comment]
Ready for re-review 👍

**Review by tearingItUp786 — APPROVED** [review]
🚀

**Pull request merged**